tests/Doc/SetsTutorial.cs

```diff
@@ -0,0 +1,199 @@
+        Assert.True(res5);
+        Assert.False(res6);
+        // REMOVE_END
+
+
+        // STEP_START sinter
+        long res7 = db.SetAdd("{bikes:racing}:france", new RedisValue[] { "bike:1", "bike:2", "bike:3" });
+        long res8 = db.SetAdd("{bikes:racing}:usa", new RedisValue[] { "bike:1", "bike:4" });
+
+        RedisValue[] res9 = db.SetCombine(SetOperation.Intersect, new RedisKey[] { "{bikes:racing}:france", "{bikes:racing}:usa" });
+        Console.WriteLine(string.Join(", ", res9)); // >>> bike:1
+        // STEP_END
+
+        // Tests for 'sinter' step.
+        // REMOVE_START
+        Assert.Equal(3, res7);
+        Assert.Equal(2, res8);
+        Assert.Equal("bike:1", string.Join(", ", res9));
+        // REMOVE_END
+
+
+        // STEP_START scard
+        long res10 = db.SetAdd("bikes:racing:france", new RedisValue[] { "bike:1", "bike:2", "bike:3" });
+        long res11 = db.SetLength("bikes:racing:france");
+        Console.WriteLine(res11);   // >>> 3
+        // STEP_END
+
+        // Tests for 'scard' step.
+        // REMOVE_START
+        Assert.Equal(3, res11);
+        delRes = db.KeyDelete("bikes:racing:france");
+        // REMOVE_END
+
+
+        // STEP_START sadd_smembers
+        long res12 = db.SetAdd("bikes:racing:france", new RedisValue[] { "bike:1", "bike:2", "bike:3" });
+        RedisValue[] res13 = db.SetMembers("bikes:racing:france");
+        Console.WriteLine(string.Join(", ", res13));    // >>> bike:3, bike:2, bike:1
+        // STEP_END
+
+        // Tests for 'sadd_smembers' step.
+        // REMOVE_START
+        Assert.Equal(3, res12);
+        // REMOVE_END
+
+
+        // STEP_START smismember
+        bool res14 = db.SetContains("bikes:racing:france", "bike:1");
+        Console.WriteLine(res14);   // >>> true
+
+        bool[] res15 = db.SetContains("bikes:racing:france", new RedisValue[] { "bike:2", "bike:3", "bike:4" });
+        Console.WriteLine(string.Join(", ", res15));    // >>> True, True, False
+        // STEP_END
+
+        // Tests for 'smismember' step.
+        // REMOVE_START
+        Assert.True(res14);
+        Assert.Equal("True, True, False", string.Join(", ", res15));
+        // REMOVE_END
+
+
+        // STEP_START sdiff
+        long res16 = db.SetAdd("{bikes:racing}:france", new RedisValue[] { "bike:1", "bike:2", "bike:3" });
+        long res17 = db.SetAdd("{bikes:racing}:usa", new RedisValue[] { "bike:1", "bike:4" });
+        RedisValue[] res18 = db.SetCombine(SetOperation.Difference, new RedisKey[] { "{bikes:racing}:france", "{bikes:racing}:usa" });
+        Console.WriteLine(string.Join(", ", res18));    // >>> bike:2, bike:3
+        // STEP_END
+
+        // Tests for 'sdiff' step.
+        // REMOVE_START
+        Assert.Equal(0, res16);
+        Assert.Equal(0, res17);
+        // REMOVE_END
+
+
+        // STEP_START multisets
+        long res19 = db.SetAdd("{bikes:racing}:france", new RedisValue[] { "bike:1", "bike:2", "bike:3" });
```

Comment:
The curly brackets here might be confusing, are they for clustered setup? In the Python snippets at https://redis.io/docs/latest/develop/data-types/sets/ there are no `{...}`.

Reply:
Yeah, they are [hashtags](https://redis.io/docs/latest/operate/rc/databases/configuration/clustering/#standard-hashing-policy). When you have a sharded database, you can't do multi-object commands (eg, using 2 sets or 2 lists) unless they both have the same hash slot. Using hashtags, only the bit between the {...} is hashed, so you ensure you get the same hash slot. I think they are not used in the Python example because the Python repo doesn't test against a multi-slot DB.

Reply:
Do you think they are relevant for these examples? For me, at first sight, they seem unnecessary. Especially that they show up in some snippets and not in others.

Reply:
They're not ideal for documentation but some of the tests fail without them.